DotNet/Submission/Listeners/SubmitReportListener.cs

```diff
@@ -406,6 +417,83 @@ await File.WriteAllTextAsync(submissionDirectory + "/" + fileName, contents,
             }
         }
 
+        protected void GenerateSubmissionMetrics(Bundle? otherResourcesBundle, List<string>? patientFilesWritten, string facilityId, DateTime startDate, DateTime endDate )
+        {
+            if(otherResourcesBundle == null) { return; }    
+            if (patientFilesWritten == null || patientFilesWritten.Count == 0) { return; }
+
+            var totalResources = otherResourcesBundle.GetResources().Count();
+            var resourceTypes = new Dictionary<string, int>();
+            var medicationCodes = new Dictionary<string, int>();
+            foreach (var patientFile in patientFilesWritten)
+            {
+                string contents = File.ReadAllText(patientFile);
+                var bundle = JsonSerializer.Deserialize<Bundle>(contents, new JsonSerializerOptions().ForFhir());
+
+                var resources = bundle.GetResources();
+
+                if (resources == null)
+                    continue;
+
+                foreach (var resource in resources)
+                {
+                    totalResources += 1;
+
+                    var resType = resource.TypeName;
+
+                    if (!resourceTypes.TryAdd(resType, 1))
+                    {
+                        resourceTypes[resType] += 1;
+                    }
+
+                    if (resource is Medication)
+                    {
+                        var med = (Medication)resource;
+                        var code = med.Code?.Coding?.FirstOrDefault();
+                        if (code != null)
+                        {
+                            string val = $"{code.Display}|{code.Code}|{code.System}";
```

Comment:
I've gone back and forth on whether or not `code.Display` should be included here. The `Display` can vary for a single `Code` from one facility to another. We are likely limited by Prometheus/OTEL in how this dynamic is handled, though. We might want to consider tracking `<display>|<code>|<system` in one metric, and `<code>|<system>` in another metric (i.e. `medicationCodesWithDisplay` vs. `medicationCodes`). On one hand, it's valuable to see _a_ `display` value when looking at these analytics, on the other hand, it could result in the same `code` being shown multiple times because facilities have different `display` representations for the same code.